bond_app/fence_token_vending.py

```diff
@@ -62,15 +62,9 @@ def get_service_account_key_json(self, user_info):
         self.cache_api.add(namespace=self.provider_name, key=user_info.id, value=key_json, expires_in=seconds_to_expire)
         return key_json
 
-    def _fetch_real_user_info(self, user_info):
-        real_user_info = self.sam_api.user_info(user_info.token)
-        if real_user_info is None:
-            raise exceptions.Unauthorized("user not found in sam")
-        return real_user_info
-
     def _get_oauth_access_token(self, provider_user):
         refresh_token = self.refresh_token_store.lookup(provider_user.user_id, self.provider_name)
         if refresh_token is None:
-            raise exceptions.BadRequest("Fence account not linked")
+            raise exceptions.NotFound("Fence account not linked. {}".format(str(provider_user)))
```

Comment:
What is the rationale for this change? I don't recall what we were thinking at the time but BadRequest must have made sense.

Reply:
I don't think BadRequest is a good match for what's going on. Bond wasn't apple to find a valid refresh token for the user, either because it expired or the user has never linked their account. There's nothing bad about the user's request. They wouldn't have to change the request if their token existed, so I don't think BadRequest is a good fit.

I'm not sure if Unauthorized, Forbidden, or NotFound is best. We didn't find the refresh token, we're not sure why. The user might just need to authorize their fence account through Bond, which sounds Unauthorized-ish to me. Or maybe the current action is Forbidden since we know who the user is in Sam, but they don't have a refresh token that matches. WDYT?